services/host_service.js

```diff
@@ -0,0 +1,57 @@
+
+const {Host} = require('../db/host')
+const {TimeSlot} = require('../db/time_slot')
+const {RestUtils} = require('../utils/rest_utils')
+const {Checker} = require('./data_checker')
+
+Host.timeSlots = Host.hasMany(TimeSlot);
+
+const HostService = {
+    createHost: async () => {
+        return Host.create({
+            email: "andrzej@duda.pl",
+            TimeSlots: [{
+                startDatetime: Date(),
+                duration: 45
+            }]
+        }, {
+            include: [{
+                association: Host.timeSlots,
+            }
+            ]
+        })
+    },
+
+    getHostWithTimeSlots: async(uuid) => {
+        const host = await Host.findOne({
+            attributes: ['email'],
+            where: {
+                uuid: uuid
+            },
+            include: TimeSlot
+        })
+        return {...RestUtils.createResponse("Hosts time slots"), host: host}
+    },
+
+    updateHostsTimeSlots: async (uuid, timeSlots) => {
+        TimeSlot.destroy({
+            where: {
+                HostUuid: uuid
+            }
+        }).then(async _ => {
+            for (const timeSlot of timeSlots) {
+                const checkData = Checker.checkTimeSlot(timeSlot)
+                if (checkData !== true) return RestUtils.createResponse(`Cannot set timeSlots in the past!`, RestUtils.BAD_REQUEST_CODE)
```

Comment:
Ten return wywali Cię na pierwszej próbie dodania terminu w przeszłości, ale chyba nie powinno to tak działać. Ogólnie albo wywala wszystkie, albo dodaje wszystkie prawilne. Jakbym miał sytuację:
1. dobry
2. dobry
3. w przeszłości
4. dobry
5. dobry
To terminy 1 i 2 się dodadzą, na trzecim się wywali, ale 4 i 5 się nie dodadzą, mimo iż były dobre. Wygląda to, jakbyśmy działali losowo. Więc albo dodajemy wszystkie na raz i mówimy, które były złe, albo nie dodajemy żadnego i prosimy, żeby ogarnął dupę.

Reply:
Ogólnie nie wiem jak inni ale rozważyłbym przeniesienie przynajmniej części zawartości pętli do osobnej funkcji